ortools/sat/csharp/CpSolver.cs

```diff
@@ -26,238 +23,175 @@ namespace Google.OrTools.Sat
  * variables in the best solution, as well as general statistics of the search.
  * </remarks>
  */
-public class CpSolver
+public sealed class CpSolver : IDisposable
 {
-    /** <summary>Solves the given model, and returns the solve status.</summary> */
-    public CpSolverStatus Solve(CpModel model, SolutionCallback cb = null)
+    private LogCallback? _log_callback;
```

Comment:
Google **C++** coding convention use `_` as suffix for private member why did you choose to use `_` as prefix ?
Is it backed by a .Net naming convention (and tooling) ? 

ref: https://google.github.io/styleguide/cppguide.html#Variable_Names

Reply:
prefix is a standard of c#. I normally use analyzer to enforce static code analyses. But it is fine. I will follow your convention to have an uniformed way 👌

Reply:
@ChangeTheCode AFAIK Google do not provide a naming convention for .Net (i.e. not an official language here) so in this case we prefer to stick with "default language/microsoft" naming convention so we can leverage ecosystem tooling -> I would prefer to keep the `_` prefix if it reduce the friction...

Reply:
> [...] I normally use analyzer to enforce static code analyses.

May I have the name of your "analyzer tools" i.e. how I can reproduce the errors you got "on my machine (TM)"

Reply:
I can create therefore, a separate PR. 
Lets focus on the disposable part